### PR TITLE
feat(formula): align to template formula

### PR DESCRIPTION
**Pull request by noelmcloughlin**
### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [ ] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [x] `[feat]`     A new feature
- [ ] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

YES

### Describe the changes you're proposing

There are about ten Jetbrains formulas in saltstack-formulas which need attention.  This formula is a template for all ten jetbrains formulas and has been migrated to align to template-formula. Travis CI tests are passing and this PR has been tested successfully on MacOS.

This PR will be followed by ten more PR so that all Jetbrains formulas are aligned and in CI/CD.

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### CODE REVIEW

**Comment by noelmcloughlin** [comment]
MacOS trace
```
local:
----------
          ID: intellij-macos-app-install-curl
    Function: file.directory
        Name: /tmp/epp/intellij
      Result: True
     Comment: Files cleaned from directory /tmp/epp/intellij
     Started: 00:14:55.436915
    Duration: 42.334 ms
     Changes:   
              ----------
              /tmp/epp/intellij/intellij-2020.1:
                  ----------
                  removed:
                      Removed due to clean
              removed:
                  - /tmp/epp/intellij/intellij-2020.1
----------
          ID: intellij-macos-app-install-curl
    Function: pkg.installed
        Name: curl
      Result: True
     Comment: All specified packages are already installed
     Started: 00:14:59.628188
    Duration: 10061.295 ms
     Changes:   
----------
          ID: intellij-macos-app-install-curl
    Function: cmd.run
        Name: curl -Lo /tmp/epp/intellij/intellij-2020.1 https://download.jetbrains.com/idea/ideaIC-2020.1.dmg
      Result: True
     Comment: Command "curl -Lo /tmp/epp/intellij/intellij-2020.1 https://download.jetbrains.com/idea/ideaIC-2020.1.dmg" run
     Started: 00:15:09.693581
    Duration: 1056668.818 ms
     Changes:   
              ----------
              pid:
                  42086
              retcode:
                  0
              stderr:
                    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                                   Dload  Upload   Total   Spent    Left  Speed
                  
                    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
                  100   138  100   138    0     0    801      0 --:--:-- --:--:-- --:--:--   802
                  
                    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
                    0  600M    0  447k    0     0   340k      0  0:30:08  0:00:01  0:30:07  451k
                    0  600M    0  895k    0     0   380k      0  0:26:55  0:00:02  0:26:53  442k
                    0  600M    0 1471k    0     0   441k      0  0:23:13  0:00:03  0:23:10  489k
              
                    << CUT >>
              
                   99  600M   99  599M    0     0   582k      0  0:17:36  0:17:35  0:00:01  800k
                   99  600M   99  600M    0     0   582k      0  0:17:36  0:17:36 --:--:--  745k
                  100  600M  100  600M    0     0   582k      0  0:17:36  0:17:36 --:--:--  753k
              stdout:
----------
          ID: intellij-macos-app-install-checksum
    Function: module.run
        Name: file.check_hash
      Result: True
     Comment: onlyif condition is false
     Started: 00:32:46.364667
    Duration: 771.576 ms
     Changes:   
----------
          ID: intellij-macos-app-install-checksum
    Function: file.absent
        Name: /tmp/epp/intellij/intellij-2020.1
      Result: True
     Comment: State was not run because onfail req did not change
     Started: 00:32:47.136970
    Duration: 0.004 ms
     Changes:   
----------
          ID: intellij-macos-app-install-macpackage
    Function: macpackage.installed
        Name: /tmp/epp/intellij/intellij-2020.1
      Result: True
     Comment: IntelliJ IDEA CE.app installed
     Started: 00:32:47.137133
    Duration: 7183.909 ms
     Changes:   
              ----------
              installed:
                  - IntelliJ IDEA CE.app
----------
          ID: intellij-macos-app-install-macpackage
    Function: file.managed
        Name: /tmp/mac_shortcut.sh
      Result: True
     Comment: File /tmp/mac_shortcut.sh is in the correct state
     Started: 00:32:54.321359
    Duration: 37.875 ms
     Changes:   
----------
          ID: intellij-macos-app-install-macpackage
    Function: cmd.run
        Name: /tmp/mac_shortcut.sh
      Result: True
     Comment: Command "/tmp/mac_shortcut.sh" run
     Started: 00:32:54.359796
    Duration: 1193.777 ms
     Changes:   
              ----------
              pid:
                  44068
              retcode:
                  0
              stderr:
              stdout:
                  alias file IntelliJ IDEA CE alias of folder Desktop of folder nmcloughlin of folder Users of startup disk
----------
          ID: intellij-config-file-file-managed-environ_file
    Function: file.managed
        Name: /etc/defaults/intellij.sh
      Result: True
     Comment: File /etc/defaults/intellij.sh is in the correct state
     Started: 00:32:55.554266
    Duration: 1634.646 ms
     Changes:   

Summary for local
------------
Succeeded: 9 (changed=4)
Failed:    0
```

**Comment by myii** [comment]
@noelmcloughlin I've noticed your `template-formula` conversion PRs and the new formulas that you've added -- I'll try to get around to reviewing/finalising them soon (i.e. GitHub settings).  Thanks for your patience.

File: kitchen.yml [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      run_command: /sbin/init
+  - name: arch-base-latest-2019-2-py2
+    driver:
+      image: netmanagers/salt-2019.2-py2:arch-base-latest
+      run_command: /usr/lib/systemd/systemd
+
+
+provisioner:
+  name: salt_solo
+  # log_level: debug
```
Comment (myii) [inline kitchen.yml]:
```suggestion
  log_level: debug
```

**Review by myii — COMMENTED** [review]
Nice conversion, @noelmcloughlin.  Only got one line being highlighted, as shown inline.  Are there issues when debug logging is used?

* Checked using https://github.com/myii/ssf-formula/pull/181

**Comment by myii** [comment]
Looking at https://github.com/saltstack-formulas/jetbrains-intellij-formula/pull/29#issuecomment-624856478, worth squashing some of the commits?

**Comment by myii** [comment]
Actually, I only did the _automated_ tests so far.  I still need to look at the `README` and the `FORMULA` file.  Bit busy right now but I'll do this when I get a chance.

**Comment by myii** [comment]
@noelmcloughlin That last commit message is getting caught by the linter.  I've checked the other two files and found that the `depth` is missing from the `README`:

```rst
.. contents:: **Table of Contents**
   :depth: 1
```

So to finish off, how about adding this last change and then squashing (some of) the commits to avoid the warnings in https://github.com/saltstack-formulas/jetbrains-intellij-formula/pull/29#issuecomment-624856478?

**Comment by noelmcloughlin** [comment]
Comments implemented
1. enable kitchen debug
2. depth=1
3. squashed commits/conflicts.

I updated all my PRs for 1&2 above - hopefully, they are good now too.  Thanks!!!

**Comment by myii** [comment]
> Comments implemented
> 
> 1. enable kitchen debug
> 2. depth=1
> 3. squashed commits/conflicts.
> 
> 
> I updated all my PRs for 1&2 above - hopefully, they are good now too. Thanks!!!

@noelmcloughlin You're welcome, we're merged here.  There was an extra blank line that crept into `kitchen.yml` but I added a tiny commit to deal with that.

**Pull request merged**